libs/blocks/path-finder/path-finder.js

```diff
@@ -0,0 +1,103 @@
+import login from '../../tools/sharepoint/login.js';
+import { account } from '../../tools/sharepoint/state.js';
+import getServiceConfig from '../../utils/service-config.js';
+import { getReqOptions } from '../../tools/sharepoint/msal.js';
+import { createTag } from '../../utils/utils.js';
+
+const SCOPES = ['files.readwrite', 'sites.readwrite.all'];
+const TELEMETRY = { application: { appName: 'Milo - Where am I' } };
+
+const getSharePointDetails = (() => {
+  let site;
+  let driveId;
+  let reqOpts;
+
+  return async () => {
+    if (site && driveId && reqOpts) return { site, driveId, reqOpts };
+    console.log('fetching sp details');
+    const { sharepoint } = await getServiceConfig(origin);
+    ({ site } = sharepoint);
+    driveId = sharepoint.driveId ? `drives/${sharepoint.driveId}` : 'drive';
+    reqOpts = getReqOptions();
+
+    return { site, driveId, reqOpts };
+  };
+})();
+
+function getItemId() {
+  const referrer = new URLSearchParams(window.location.search).get('referrer');
+  const sourceDoc = referrer?.match(/sourcedoc=([^&]+)/)[1];
+  const sourceId = decodeURIComponent(sourceDoc);
+  return sourceId.slice(1, -1);
+}
+
+async function linkToPage(e, form, input, target) {
+  e.preventDefault();
+
+  const { site, driveId, reqOpts } = await getSharePointDetails();
+  const resp = await fetch(`${site}/${driveId}/root:${input.value}`, reqOpts);
+  if (!resp.ok) {
+    form.classList.add('error');
+    return;
+  }
+
+  form.classList.remove('error');
+  const json = await resp.json();
+  window.open(json.webUrl, target || '_blank');
+}
+
+function buildUi(el, path) {
+  const form = createTag('form');
+  const input = createTag('input', { type: 'text', value: path });
+  const btn = createTag('button', { }, 'Go');
+  const jumpBtn = createTag('button', { }, '');
+  jumpBtn.classList.add('new-tab');
+
+  form.addEventListener('submit', (e) => linkToPage(e, form, input, '_parent'));
+  jumpBtn.addEventListener('click', (e) => linkToPage(e, form, input));
+  form.append(input, btn, jumpBtn);
+  el.append(form);
+}
+
+async function setup(el) {
+  await login({ scopes: SCOPES, telemetry: TELEMETRY });
+  if (!account.value.username) {
+    window.lana.log('Could not login to MS Graph');
```

Comment:
@robert-bogos has recently done some work on categorizing LANA messages, maybe he has some suggestions for this one as well

Reply:
Yes, we should add tags to LANA logs to facilitate filtering them in Splunk.
For this particular log I would go for 
```suggestion
    window.lana.log('Could not login to MS Graph', { tags: 'errorType=info,module=path-finder' });
```
You can see more examples and details about these tags in [here](https://github.com/adobecom/milo/pull/2314).

Reply:
Thanks @robert-bogos. Updated.